pytumblr/__init__.py

```diff
@@ -163,7 +166,561 @@ def blog_following(self, blogname, **kwargs):
+        if post_type == 'text':
+            valid += ['title', 'body']
+        elif post_type == 'photo':
+            valid += ['caption', 'link', 'source', 'data', 'photoset_layout']
+        elif post_type == 'quote':
+            valid += ['quote', 'source']
+        elif post_type == 'link':
+            valid += ['title', 'url', 'description', 'thumbnail']
+        elif post_type == 'chat':
+            valid += ['title', 'conversation']
+        elif post_type == 'audio':
+            valid += ['caption', 'external_url', 'data']
+        elif post_type == 'video':
+            valid += ['caption', 'embed', 'data']
+
+        return valid
+
+    def _send_post(self, blogname, params):
+        """
+        Formats parameters and sends the API request off. Validates
+        common and per-post-type parameters and formats your tags for you.
+
+        :param blogname: a string, the blogname of the blog you are posting to
+        :param params: a dict, the key-value of the parameters for the api request
+        :param valid_options: a list of valid options that the request allows
+
+        :returns: a dict parsed from the JSON response
+        """
+        url = "/v2/blog/{0}/post".format(blogname)
+        valid_options = self._post_valid_options(params.get('type', None))
+
+        if len(params.get("tags", [])) > 0:
+            # Take a list of tags and make them acceptable for upload
+            params['tags'] = ",".join(params['tags'])
+
+        return self.send_api_request("post", url, params, valid_options)
+
+    def send_api_request(self, method, url, params={}, valid_parameters=[], needs_api_key=False):
+        """
+        Sends the url with parameters to the requested url, validating them
+        to make sure that they are what we expect to have passed to us
+
+        :param method: a string, the request method you want to make
+        :param params: a dict, the parameters used for the API request
+        :param valid_parameters: a list, the list of valid parameters
+        :param needs_api_key: a boolean, whether or not your request needs an api key injected
+
+        :returns: a dict parsed from the JSON response
+        """
+        if needs_api_key:
+            params.update({'api_key': self.request.consumer_key})
+            valid_parameters.append('api_key')
+
+        files = {}
+        if 'data' in params:
+            if isinstance(params['data'], list):
+                for idx, data in enumerate(params['data']):
+                    files['data['+str(idx)+']'] =  open(params['data'][idx], 'rb')
+            else:
+                files = {'data': open(params['data'], 'rb')}
+            del params['data']
+
+        validate_params(valid_parameters, params)
+        if method == "get":
+            return self.request.get(url, params)
+        else:
+            return self.request.post(url, params, files)
+
+
+# OAuth2 Client
+
+class TumblrRestClient2(object):
```

Comment:
This class is almost 100% duplicated from the TumblrRestClient OAuth1 class,
I think it would be better to update that class to create a `TumblrRequest` or `TumblrRequest2` based on the input params, then any users of pytumblr just have to change the params in order to use OAuth2 when OAuth1 is eventually deprecated.
An alternative to changing the existing __init__ would be to add a classmethod that acts as an alternative constructor